custom_model_runner/datarobot_drum/drum/args_parser.py

```diff
@@ -683,10 +683,24 @@ def _reg_arg_dr_api_access(*parsers):
     @staticmethod
     def _reg_arg_triton_server_access(*parsers):
         for parser in parsers:
-            parser.add_argument(
+            server_group = parser.add_mutually_exclusive_group(required=False)
+            server_group.add_argument(
```

Comment:
Probably should be organized as a single param with multiple choices.
--with-nvidia-server=triton/nemo

Reply:
Especially when you add mutually exclusive.
Could you redo this in this PR?

Reply:
We plan to add at least one more server too though: e.g. vLLM which has no relation to nvidia, so maybe `--with-nvidia-server` needs to be more generic: `--with-inference-server` ?

Reply:
Or even more general: add a param to specify predictor instead of resolving it by language/artifacts.
--predictor==Python/R/Java/Triton/Nemo.

Reply:
Maybe it worth introducing `--predictor` param. If specified, ignore language-based resolution and just used what is provided

Reply:
🤔 but from inside custom models workshop... customers don't have control over CLI flags right? So we kinda need to auto-detect stuff? Or are we saying that the env we are building will be locked to a specific predictor? I'll have to defer to Alex on that (I'm not sure if that model works with Triton-python env)

Reply:
Most of the CLI params are also easily duplicated with ENV vars, we have this mechanism built in into DRUM.

By default, our dropin envs have only deps that are reflected in the env names: sklearn, pytorch. So if you upload onnx artifact into sklearn env it likley won't work. Though you can still install additional packages and it will.

So you are right about locking, maybe not everything should be locked.
But in the PR Alex tries to resolve predictor based on param provided and target type, which seems not right. It would be better to provide predictor and verify, that target type is suitable to that predictor.

Reply:
> Most of the CLI params are also easily duplicated with ENV vars, we have this mechanism built in into DRUM

How do our users set env vars though? (other than via runtime params but users shouldn't actually know those are env vars as it is an implementation detail).

Otherwise I think Alex will have to explain his vision for how he wants envs to work with these new inference servers. I think the Triton-py3 env is most complicated because we want a single env that has Triton-py3 installed but Triton can handle Pytorch, Onnx, Tensorflow, etc all from this one env so maybe that's where things are not aligning perfectly?

Reply:
> customers don't have control over CLI flags right? So we kinda need to auto-detect stuff based on the files we see?

We don't need to autodetect artifacts. User selects a correct drop-in environment, where the `start-server.sh` sets the `--predictor=triton` or `--predictor=nemo` flag.

Reply:
> Or even more general: add a param to specify predictor instead of resolving it by language/artifacts.
--predictor==Python/R/Java/Triton/Nemo.

@elatt @yakov-g I'm going to add the `--gpu-predictor` param and it will have only the `triton/nemo/vllm` choices. I know, Triton may work without the gpu, but the main reason we adding it - is the GPU support.

Reply:
@yakov-g I've added the `--gpu-predictor=triton|nemo|vllm` option. If you think it would better to have the generic `--predictor` option, I'll add the change it in the followup.